content/error-handling/error-codes.textile

```diff
@@ -0,0 +1,778 @@
+
+*Resolution*: The following steps can help resolve this issue:
+* Ensure that the @_ablyChannel@ field is present at the root level of the change event.
+* Avoid nesting it inside other sections, such as @$fullDocument._ablyChannel@.
+* The field must be part of the main structure of the change event to allow proper identification.
+
+h3(#72005). 72005: Ingress invalid pipeline
+
+This error occurs when the "MongoDB Change Stream":https://www.mongodb.com/docs/manual/changeStreams/#change-streams fails to start due to an invalid pipeline. An example of this error is when the pipeline syntax does not conform to MongoDB's requirements or contains unrecognized operators.
+
+*Resolution*: The following steps can help resolve this issue:
+* Review the pipeline syntax for errors and ensure all operators are valid.
+* Adjust the pipeline to match MongoDB's accepted structure and syntax guidelines.
+
+h3(#72006). 72006: Unable to resume from change stream
+
+This error occurs when the "MongoDB Change Stream":https://www.mongodb.com/docs/manual/changeStreams/#change-streams cannot be resumed because the resume token document stored in MongoDB is not in the correct format.
+
+*Resolution*: The following steps can help resolve this issue:
+* Verify the format of the stored resume token document in MongoDB.
+* Ensure the token meets the expected structure and format required for MongoDB Change Stream resumption.
+* Refer to the MongoDB documentation for guidelines on properly storing and using resume tokens.
+
+h3(#72007). 72007: Unable to store change stream resume token
+
+This error occurs when the "MongoDB Change Stream":https://www.mongodb.com/docs/manual/changeStreams/#change-streams resume token cannot be stored in the @ably@ collection of the database.
+
+*Resolution*: The following steps can help resolve this issue:
+* Verify that the integration rule and MongoDB connection string are correctly configured.
+* Ensure the database user has the necessary read and write permissions for the @ably@ collection.
+* Adjust permissions if needed to allow the token to be successfully stored.
+
+h2(#push). Push notification errors
+
+h3(#40015). 40015: Invalid device ID
+
+This error occurs when the ID property in a "DeviceDetails":/docs/api/rest-sdk/push-admin#device-details object, sent during a push notification registration request, is invalid. The ID must be a string.
+
+h3(#61002). 61002: Activation failed: Present clientId is not compatible with existing device registration
+
+This error occurs when you previously "activated":/configure/device#activate-devices a device for push notifications with a specific @clientId@, but then changed the @clientId@ used for authentication. The mismatch causes the error because the push notification setup tracks the @clientId@ and other details to prevent accidental changes between app launches. The @clientId@ is linked to registrations, such as subscribing by @clientId@.
```

Comment:
```suggestion
This error occurs when you previously "activated":/docs/push/configure/device#activate-devices a device for push notifications with a specific @clientId@, but then changed the @clientId@ used for authentication. The mismatch causes the error because the push notification setup tracks the @clientId@ and other details to prevent accidental changes between app launches. The @clientId@ is linked to registrations, such as subscribing by @clientId@.
```